src/vm/checker/errors.rs

```diff
@@ -124,3 +127,80 @@ impl error::Error for CheckError {
         }
     }
 }
+
+fn formatted_expected_types(expected_types: & Vec<TypeSignature>) -> String {
+    let mut expected_types_joined = String::new();
+    expected_types_joined = format!("'{}'", expected_types[0]);
+
+    if expected_types.len() > 2 {
+        for expected_type in expected_types[1..expected_types.len()-2].into_iter() {
+            expected_types_joined.push_str(&format!(", '{}'", expected_type));
+        }
+    }
+    expected_types_joined.push_str(&format!(" or '{}'", expected_types[expected_types.len()-1]));
+    expected_types_joined
+}
+
+impl DiagnosableError for CheckErrors {
+
+    fn message(&self) -> String {
+        match &self {
+            CheckErrors::UnknownListConstructionFailure => format!("invalid syntax for list definition"),
+            CheckErrors::ListTypesMustMatch => format!("expecting elements of same type in a list"),
+            CheckErrors::ConstructedListTooLarge => format!("reached limit of elements in a list"),
+            CheckErrors::TypeError(expected_type, found_type) => format!("expecting expression of type '{}', found '{}'", expected_type, found_type),
+            CheckErrors::UnionTypeError(expected_types, found_type) => format!("expecting expression of type {}, found '{}'", formatted_expected_types(expected_types), found_type),
+            CheckErrors::ExpectedOptionalType(found_type) => format!("expecting expression of type 'optional', found '{}'", found_type),
+            CheckErrors::ExpectedResponseType(found_type) => format!("expecting expression of type 'response', found '{}'", found_type),
+            CheckErrors::CouldNotDetermineResponseOkType => format!("expecting a response of type 'ok'"),
```

Comment:
These two errors are a bit strange (`CouldNotDetermineResponse...`).

They occur in cases where a function returns a Response type, but only ever returns an `ok` or an `er`. For example:

```
(define (foo) (ok 1))
```

Is totally legal, but if you try to do something like:

```
(expects-err! (foo) 'false)
```

You cannot typecheck _that_ statement, which produces the type error. So maybe something along the lines of:

"attempted to obtain 'ok' value from response, but 'ok' type is indeterminate" ?

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1041/commits/b7bb64d185dba5d8c943f494df68638ab2a5e04c